src/azul/chalice.py

```diff
@@ -400,6 +404,41 @@ def catalog(self) -> str:
     def _controller(self, controller_cls: Type[C], **kwargs) -> C:
         return controller_cls(app=self, **kwargs)
 
+    def render_swagger_ui(self, swagger_ui_template: str) -> str:
+        base_url = self.base_url
+        redirect_url = furl(base_url).add(path='oauth2_redirect')
+        deployment_url = furl(base_url).add(path='openapi')
+        return chevron.render(swagger_ui_template, {
+            'DEPLOYMENT_PATH': json.dumps(str(deployment_url.path)),
+            'OAUTH2_CLIENT_ID': json.dumps(config.google_oauth2_client_id),
+            'OAUTH2_REDIRECT_URL': json.dumps(str(redirect_url)),
+            'NON_INTERACTIVE_METHODS': json.dumps([
+                f'{path}/{method.lower()}'
+                for path, method in self.non_interactive_routes
+            ])
+        })
+
+    def swagger_ui(self) -> Response:
+        swagger_ui_html = self.load_static_resource('swagger-ui.html')
+        return Response(status_code=200,
+                        headers={'Content-Type': 'text/html'},
+                        body=swagger_ui_html)
+
+    _static_swagger_resources = {
+        'swagger-ui.css',
```

Comment:
The reason they are mentioned here is for security reasons so that this endpoint can't be abused to read arbitrary files. There may be a better way to achieve that.